New version: Nushell.Nushell version 0.93.0

fdncred opened the pull request:
### Pull request has been created with [WinGet Releaser v2](https://github.com/vedantmgoyal2009/winget-releaser) :rocket:
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/151729)

fdncred committed:
New version: Nushell.Nushell version 0.93.0

fdncred committed:
let's try again

fdncred commented:
These errors are a mistake on our part. We'll be fixing this shortly.

stephengillie commented:
Automatic Validation ended with:

> Installation failed with exit code -1978335226
> 2024-04-30 17:11:49.491 [CLI ] ShellExecute installer failed: 1603
> CAQuietExec:  Error:   [31mÃ—[0m pipeline input and output are the same file
> CAQuietExec:  Error 0x80070001: Command line returned an error.
> CAQuietExec:  Error 0x80070001: QuietExec Failed
> CAQuietExec:  Error 0x80070001: Failed in ExecCommon method
> CustomAction ReplacePathsInWindowsTerminalProfile returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)
> Property(S): ErrorDialog = ErrorDlg
> MSI (s) (30:4C) [17:11:49:257]: Product: nu -- Installation failed.

Automated error analysis suggests 1603 may mean Install successful but system restart may be required.

Automated error analysis suggests -1978335226 may mean Install successful but system restart may be required.

(Automated response - build 878.)

devyn commented on manifests/n/Nushell/Nushell/0.93.0/Nushell.Nushell.installer.yaml:
```diff
@@ -0,0 +1,30 @@
+Installers:
+- Architecture: x64
+  InstallerUrl: https://github.com/nushell/nushell/releases/download/0.93.0/nu-0.93.0-x86_64-pc-windows-msvc.msi
+  InstallerSha256: 23432A484E15A1116E55DCD7FBAE4BE52CDD7428E2BC880FC266EE3AC6F2687D
```
new: `72043efb23307476849edeca75a0f5b01faf872deedff2d3b22272b609ddd649`

fdncred replied:
updated

devyn commented on manifests/n/Nushell/Nushell/0.93.0/Nushell.Nushell.installer.yaml:
```diff
@@ -0,0 +1,30 @@
+  ProductCode: '{3F10AD17-1EA0-4037-8EDB-9BDEB9BE6A2F}'
+- Architecture: arm64
+  InstallerUrl: https://github.com/nushell/nushell/releases/download/0.93.0/nu-0.93.0-aarch64-pc-windows-msvc.msi
+  InstallerSha256: 1BC69DFC0EC3015D032ABDC653CF87A14A030FB2BC1DD53793240982116C9678
```
new: `6d8db18a24591ee516a47dedcf67ba09c12df16292ef0604bb0342d0505ec911`

fdncred replied:
updated

fdncred committed:
update hashes

fdncred commented:
@stephengillie 

Yet another failure. I ran it in the sandbox and it failed with
```
Action start 7:27:19: InstallFinalize.
CAQuietExec:  Error 0xc0000135: Command line returned an error.
CAQuietExec:  Error 0xc0000135: QuietExec Failed
CAQuietExec:  Error 0xc0000135: Failed in ExecCommon method
CustomAction ReplacePathsInWindowsTerminalProfile returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)
Action ended 7:27:20: InstallFinalize. Return value 3.
```
I ran it on my system and it installs without error. There is a UAC acknowledgement dialog that is presented though.

How can we get this through?

stephengillie commented:
Automatic Validation ended with:

> Installation failed with exit code -1978335226
> 2024-04-30 19:35:54.301 [CLI ] ShellExecute installer failed: 1603
> CAQuietExec:  Error 0xc0000135: Command line returned an error.
> CAQuietExec:  Error 0xc0000135: QuietExec Failed
> CAQuietExec:  Error 0xc0000135: Failed in ExecCommon method
> CustomAction ReplacePathsInWindowsTerminalProfile returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)
> Property(S): ErrorDialog = ErrorDlg
> MSI (s) (78:A4) [19:35:53:931]: Product: nu -- Installation failed.

Automated error analysis suggests 1603 may mean Install successful but system restart may be required.

Automated error analysis suggests -1978335226 may mean Install successful but system restart may be required.

(Automated response - build 880.)

fdncred commented:
Seems like we have 2 options.
1. Forget about using winget because it's such a pain and fails with nearly every release.
2. Try to make the installer perUser versus perMachine, which is not really what we want to do but winget is being a diva about it so maybe it'll work? Who knows?

stephengillie commented:
Manual Validation ended with:
> 
> 2024-05-01 13:21:36.003 [CLI ] Package hash verification failed. SHA256 in manifest [72043efb23307476849edeca75a0f5b01faf872deedff2d3b22272b609ddd649] does not match download [4d4b65f3169d4fd0cc32f6a7fcd0ad0d2cba9162c206ae050568a6f40d956f65]

(Automated response - build 880.)

fdncred committed:
update hashes

fdncred commented:
We have now changed the msi's so they are perUser installs. Hopefully this fixes the problem.

> Manual Validation ended with:
> 
> 2024-05-01 13:21:36.003 [CLI ] Package hash verification failed. SHA256 in manifest [72043efb23307476849edeca75a0f5b01faf872deedff2d3b22272b609ddd649] > does not match download [4d4b65f3169d4fd0cc32f6a7fcd0ad0d2cba9162c206ae050568a6f40d956f65]
> 
> (Automated response - build 880.)

The mismatched has was because we changed the msi's based on what I just said and you tested the new uploaded version before I had a chance to change the hash in the yaml. lol
